perlin.go

```diff
@@ -18,9 +18,11 @@ const (
 
 // Perlin is the noise generator
 type Perlin struct {
-	alpha float64
-	beta  float64
-	n     int
+	alpha  float64
+	beta   float64
+	n      int
+	source rand.Source
```

Comment:
Is there a reason to preserve `source` and `rand` in the struct as they are used only during the initialization?